content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -751,26 +751,154 @@ with the response:
-  - Must show: 'self.field' selection, has() field presence checking, 'self[key]' map access and
-    'key in self' map containment, 'self[i]' list access, all/exists/filter and how they apply to
-    maps and lists. Show more of the functions than covered in types_jsonprops.go. Provide links to
-    functions and macros in spec.  Provide link to strings extension library in cel-go that we have
-    enabled. Explain that this is an extension library.
-- Include examples table from KEP? Probably just past it in and add context.
-- Explain access to type and object meta.
-- Explain openapiv3 -> CEL declarations type mapping and include the table from the KEP. Link to OpenAPIv3 and CEL documentation about types.
-- Explain int-or-string, preserve-unknown, nullable, embedded. Provide some short examples.
-- Explain escaping using a table. Provide some short examples. Provide guidance on how to name
-  properties (both in this section of this document and elsewhere in this document where property
-  names are introduced/discussed).
-- Explain + and == for list maps and list sets (table? whatever looks better)
-- DO NOT: provide all the motivation and design rationale from the KEP.
+* spec: Invalid value: map[string]interface {}{"maxReplicas":10, "minReplicas":0, "replicas":20}: failed rule: self.replicas <= self.maxReplicas
+```
+
+`x-kubernetes-validations` could have multiple rules. 
+
+The `rule` under `x-kubernetes-validations` represents the expression which will be evaluated by CEL.
+
+The `message` represents the message displayed when validation fails. The message is required if the Rule contains
+line breaks. The message must not contain line breaks. If unset, the message is "failed rule: {Rule}".
+
+Validation rules are compiled when CRDs are created/updated. 
+Compilation error will be returned even feature gate is disabled.
+
+Example:
+
+| Rule                                     |      Message                                                                                                                                                                         |
+| -------------                            | ------------                                                                                                                                                                         |
+|"self == true" against integer type       | Invalid value: apiextensions.ValidationRule{Rule:"self == true", Message:""}: compilation failed: ERROR: \<input>:1:6: found no matching overload for '_==_' applied to '(int, bool)'|
+|"self.nonexistingValue > 0"               | Invalid value: apiextensions.ValidationRule{Rule:"self.nonexistingValue > 0", Message:""}: compilation failed: ERROR: \<input>:1:5: undefined field 'nonexistingValue'                |
+
+The Rule is scoped to the location of the `x-kubernetes-validations` extension in the schema.
+
+The `self` variable in the CEL expression is bound to the scoped value.
```

Comment:
Put up higher closer to one of the first examples? This is one of the first things people need to understand.

Reply:
Moved up